tensorflow/core/kernels/data/concatenate_dataset_op_test.cc

```diff
@@ -536,6 +536,8 @@ TEST_P(ParameterizedConcatenateDatasetOpTest, Roundtrip) {
     TF_EXPECT_OK(iterator->Save(serialization_ctx.get(), &writer));
     TF_EXPECT_OK(writer.Flush());
     VariantTensorDataReader reader(&data);
+    TF_ASSERT_OK(concatenate_dataset->MakeIterator(iterator_ctx.get(),
```

Comment:
Please a test utility (in the test base) for restoring an iterator that makes sure the iterator is reset and it its docstring explain why it is needed. Then apply this utility everywhere (instead of the current error prone pattern that expect that the test writer understand they need to call MakeIterator before Restore). Thanks.